src/services/projectService.ts

```diff
@@ -100,6 +103,48 @@ export default class ProjectService implements IProjectService {
         return (duplicateProjects !== undefined);
     }
 
+    public async deleteTag(project: IProject,
+                           tag: string, currentAsset: IAssetMetadata): Promise<IAssetMetadata> {
+        return await this.updateProjectTags(
+            project, tag, currentAsset, (tags) => tags.filter((t) => t !== tag));
+    }
+
+    public async updateTag(project: IProject, tag: string,
+                           newTag: string, currentAsset: IAssetMetadata): Promise<IAssetMetadata> {
+        return await this.updateProjectTags(
+            project, tag, currentAsset, (tags) => tags.map((t) => (t === tag) ? newTag : t));
```

Comment:
I would break out the transformer into separate variable to make it easier to understand what is going on.